src/os/layer/preset/presetmenubutton.js

```diff
@@ -245,9 +282,60 @@ class Controller extends MenuButtonCtrl {
    * Delete the Preset from its source service
    */
   remove() {
+    // clone existing preset then update clone with latest settings
     const preset = this.scope['preset'];
+    const prompt = `<span>Permanently delete Preset "${preset.label}"? This cannot be undone.</span>`;
+
+    // As a future improvement, overwrite the result in the promise.all and flag the affeted FA's in
+    // the ImportActionManager as 'Presets'
+
+    // Refresh the application
+    ConfirmUI.launchConfirm(/** @type {osx.window.ConfirmOptions} */ ({
+      yesText: 'Delete',
+      yesIcon: 'fa fa-trash-o',
+      yesButtonClass: 'btn-danger',
+      confirm: (() => {
+        // get PresetService(s) that support remove()
+        this.getService(OsLayerPreset.PresetServiceAction.REMOVE).then(
+            (service) => {
+              this['thinking'] = true;
```

Comment:
Is the `this` context on this function correct? It doesn't look like the context argument was provided to the `.then()` call.

Reply:
:point_up: Magic. I had moved one of these then's into a function at one point during these changes and discovered those ones needed the .bind(this), but these arrow functions are OK.